openmoltools/amber.py

```diff
@@ -86,23 +89,43 @@ def build_mixture_prmtop(mol2_filenames, frcmod_filenames, box_filename, prmtop_
 
     if len(all_names) != len(mol2_filenames):
         raise(ValueError("Must have UNIQUE residue names in each mol2 file."))
+    if len(mol2_filenames) != len(frcmod_filenames):
+        raise(ValueError("Must provide an equal number of frcmod and mol2 file names."))    
+
+    #Get number of files
+    nfiles = len(mol2_filenames)
+
+    #Make temporary, hardcoded filenames for mol2 and frcmod input to avoid tleap filename restrictions
+    tmp_mol2_filenames = [ 'in%d.mol2' % n for n in range(nfiles) ]
+    tmp_frcmod_filenames = [ 'in%d.frcmod' % n for n in range(nfiles) ]
+
+    #Make temporary, hardcoded filenames for output files to avoid tleap filename restrictions
+    tmp_prmtop_filename = 'out.prmtop'
+    tmp_inpcrd_filename = 'out.inpcrd'
+
+    #Create temporary directory for working in/output
+    startdir = os.getcwd()
+    outdir = tempfile.mkdtemp() 
+ 
+    #Copy input files to temporary file names in target directory
+    for (infile, outfile) in ( mol2_filenames+frcmod_filenames, tmp_mol2_filenames+tmp_frcmod_filenames ):
```

Comment:
Yes, fixing. Not caught by Travis-CI tests because this test is skipped due to absence of `packmol`.